src/Build.OM.UnitTests/Construction/SolutionFile_Tests.cs

```diff
@@ -1067,6 +1067,60 @@ public void ParseProjectConfigurationsInSolutionConfigurations2()
             Assert.Equal(".NET", solution.GetDefaultPlatformName()); // "Default solution platform"
         }
 
+        /// <summary>
+        /// Parse solution file with comments
+        /// </summary>
+        [Fact]
+        public void ParseSolutionWithComments()
+        {
+            string solutionFileContent = @"
```

Comment:
nit:
It'd probably be good to have some comments as not the second line in a section, since it should handle comments anywhere.

Reply:
Updated test, so comment is on every other line.